File: dump_knowledge.py
import os
import PyPDF2

input_dir = 'Support-Info-Documents'
output_file = 'knowledge_base_dump.txt'

files = os.listdir(input_dir)

with open(output_file, 'w', encoding='utf-8') as out_f:
    for filename in files:
        filepath = os.path.join(input_dir, filename)
        out_f.write(f"\n\n{'='*50}\n")
        out_f.write(f"FILE: {filename}\n")
        out_f.write(f"{'='*50}\n\n")
        
        try:
            if filename.endswith('.pdf'):
                with open(filepath, 'rb') as f:
                    reader = PyPDF2.PdfReader(f)
                    for page in reader.pages:
                        text = page.extract_text()
                        if text:
                            out_f.write(text + "\n")
            else:
                with open(filepath, 'r', encoding='utf-8') as f:
                    out_f.write(f.read() + "\n")
        except Exception as e:
            out_f.write(f"ERROR READING FILE: {e}\n")

print('Dumped successfully.')

File: dump_knowledge.js
const fs = require('fs');
const path = require('path');
const pdf = require('pdf-parse');

const dir = path.join(__dirname, 'Support-Info-Documents');
const output = path.join(__dirname, 'knowledge_base_dump.txt');
const files = fs.readdirSync(dir);

let allText = '';

async function processFiles() {
    for (const file of files) {
        const filePath = path.join(dir, file);
        allText += `\n\n======================================================\n`;
        allText += `FILE: ${file}\n`;
        allText += `======================================================\n\n`;

        if (file.endsWith('.pdf')) {
            const dataBuffer = fs.readFileSync(filePath);
            try {
                const data = await pdf(dataBuffer);
                allText += data.text;
            } catch(e) {
                allText += `Error reading PDF: ${e.message}`;
            }
        } else if (file.endsWith('.html') || file.endsWith('.md') || file.endsWith('.txt')) {
            allText += fs.readFileSync(filePath, 'utf8');
        } else if (file === '7.3-Tech-Lead-Software-Architecture') {
             // Treat as text
             allText += fs.readFileSync(filePath, 'utf8');
        }
    }

    fs.writeFileSync(output, allText, 'utf8');
    console.log('Knowledge base dumped successfully.');
}

processFiles();
